src/utils/data-source.test.js

```diff
@@ -1,6 +1,61 @@
-import getDataSource from './data-source';
+import getPipelineData, { getSourceID, getDataValue } from './data-source';
+import animals from './data/animals.mock';
+import demo from './data/demo.mock';
 
-describe('getDataSource', () => {
+describe('getSourceID', () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = { ...OLD_ENV };
+    delete process.env.REACT_APP_DATA_SOURCE;
+  });
+
+  afterEach(() => {
+    process.env = OLD_ENV;
+  });
+
+  it("should return 'json' by default if no source is supplied", () => {
+    expect(getSourceID()).toBe('json');
+  });
+
+  it('should return the given datasource if set via environment variable', () => {
+    process.env.REACT_APP_DATA_SOURCE = 'animals';
+    expect(getSourceID()).toEqual('animals');
+    process.env.REACT_APP_DATA_SOURCE = 'demo';
+    expect(getSourceID()).toEqual('demo');
+  });
+});
+
+describe('getDataValue', () => {
+  it('should return the correct dataset when passed a dataset string', () => {
+    expect(getDataValue('animals')).toEqual(animals);
+    expect(getDataValue('demo')).toEqual(demo);
+  });
+
+  it("should return the string 'json' when passed 'json'", () => {
+    expect(getDataValue('json')).toEqual('json');
+  });
+
+  it("should return random data when passed 'random'", () => {
```

Comment:
I think this should probably say `"should return a dataset object when passed 'random'"` since this check seems to only be about the type.